pkg/issuer/acme/dns/dns.go

```diff
@@ -292,6 +292,30 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1.GenericIssuer
 		}
 	case providerConfig.Route53 != nil:
 		dbg.Info("preparing to create Route53 provider")
+
+		// For route53, you must specify either an AccessKeyID or a secret reference to an AccessKeyID
+		// but not both.
+		if providerConfig.Route53.AccessKeyID != "" && providerConfig.Route53.SecretAccessKeyID.Name != "" {
+			return nil, nil, fmt.Errorf("route53 accessKeyID and accessKeyIDSecretRef cannot both be specified")
+		}
+
+		// Default to the AccessKeyID literal in the configuration
+		secretAccessKeyID := strings.TrimSpace(providerConfig.Route53.AccessKeyID)
+
+		// If the AccessKeyID secret reference option is defined, override the secretAccessKeyID variable
+		if providerConfig.Route53.SecretAccessKeyID.Name != "" {
+			secretAccessKeyIDSecret, err := s.secretLister.Secrets(resourceNamespace).Get(providerConfig.Route53.SecretAccessKeyID.Name)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error getting route53 secret access key id: %s", err)
+			}
+
+			secretAccessKeyIDBytes, ok := secretAccessKeyIDSecret.Data[providerConfig.Route53.SecretAccessKeyID.Key]
```

Comment:
If `SecretAccessKeyID.Key` is not set (i.e. the user didn't set it in the API request), we should fail a lot earlier than this.

Can we add additional validation to ensure this `.Key` field _is_ set if the `.Name` field is specified (and vice-versa?).

This could either be done by defining a new struct type other than `SecretKeySelector` in the APIs package, and using a `// +kubebuilder:default` marker on the `Key` field, or otherwise using a defaulting function in this file: https://github.com/cert-manager/cert-manager/blob/c16b3cca7b418ba0d0b2bf1066514b8762984517/internal/apis/certmanager/v1/defaults.go

Reply:
We should also check here prior to using `.Key` that it *is* non-empty, and fail with a more specific/clear error in that case :)